### DISCUSSION TITLE
stm32 qspi in memory-mapped mode with dual quad-NOR on stm32H7 disco kits

### DISCUSSION THREAD

**Discussion by FRASTM** [zephyrproject-rtos/zephyr#72429] [Q&A]
Taking the https://github.com/zephyrproject-rtos/zephyr/pull/72339 to run the samples/application_development/code_relocation_nocopy/  on stm32h750b_dk and stm32h747i_disco_m7

Both targets have two qspi banks connected to 2 identical quad-spi NOR flash memories  (mt25ql512ab)
![image](https://github.com/zephyrproject-rtos/zephyr/assets/51417392/c1a1e427-fbea-44a0-8375-9d7181b72d78)

**Comment by FRASTM** [comment]
Running the sample on the stm32h750b_dk. 
The Dual-flash mode must be configured and the west flash must download the piece of _function_in_ext_flash_ directly into the quad-NOR memory.

The STM32CubeProgrammer shows that 
![Screenshot from 2024-05-07 17-20-40](https://github.com/zephyrproject-rtos/zephyr/assets/51417392/0cdd2b1c-0f6d-40d0-ace0-dbe1867fe3a8)

and  memory-mapped examination 
```
0x90000000 <function_in_ext_flash>:	0x49074a06	0xb5084807	0xf812f000	0x48074906
0x90000010 <function_in_ext_flash+16>:	0xe8bd680a	0xf0004008	0xbf00b80b	0x90000001
0x90000020 <function_in_ext_flash+32>:	0x08009dde	0x08009da6	0x240000e0	0x08009db8
```
are similar and samples execution is correct : 


```
*** Booting Zephyr OS build v3.6.0-3069-gb4c50fa37515 ***                                                                                              
Address of main function 0x800066d                                                                                                                     
Address of function_in_ext_flash 0x90000001                                                                                                            
Address of var_ext_sram_data 0x240000e0 (10)                                                                                                           
Address of function_in_sram 0x24000001                                                                                                                 
Address of var_sram_data 0x240000e4 (10)                                                                                                               
Hello World! stm32h750b_dk

```

**Comment by FRASTM** [comment]
However this is still not enough to make the **stm32h747_disco_m7** work correctly.
Even if the STM32CubeProgrammer  shows code in the 0x90000000 area, the execution fails on wrong instruction. 
--> bytes are not similar



![Screenshot from 2024-05-07 16-46-51](https://github.com/zephyrproject-rtos/zephyr/assets/51417392/1444b6da-2ff5-4d2a-8121-821c922d2f5a)

**Reply by arbrauns** [reply]
The bytes *are* similar, are you sure the QSPI clock isn't too fast? I saw similar things in my testing when the flash was being overclocked.

**Comment by FRASTM** [comment]
> The bytes _are_ similar, are you sure the QSPI clock isn't too fast? I saw similar things in my testing when the flash was being overclocked.

Reducing the `qspi-max-frequency = <8000000>; ` does not change the problem
